src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -285,6 +288,9 @@ public void delete(long exerciseId, boolean deleteStudentReposBuildPlans, boolea
         // make sure tutor participations are deleted before the exercise is deleted
         tutorParticipationRepository.deleteAllByAssessedExerciseId(exercise.getId());
 
+        // delete exercise in all achievements
+        achievementService.findAllByExerciseId(exercise.getId()).forEach(achievement -> achievement.setExercise(null));
```

Comment:
Don't you have to save the altered achievements?

Reply:
yes, I should